Lib/test/test_capi/test_misc.py

```diff
@@ -1797,6 +1797,7 @@ def test_py_config_isoloated_per_interpreter(self):
         # double checked at the time this test was written.
         config = _testinternalcapi.get_config()
         config['int_max_str_digits'] = 55555
+        config['parse_argv'] = 0
```

Comment:
Is there anything else in the config that this should consider setting? What if we add something in the future that tests like this one should set? Is there a better way to create a config for a test?

Reply:
parse_argv=0 is a workaround for the only issue that I'm aware of.

set_config() is an experimental API that I wrote, and it doesn't work well as you can see :-( I wrote a whole PEP to replace it with a better public API: PyConfig_Set(), see: https://peps.python.org/pep-0741/#get-and-set-the-runtime-configuration

set_config() sets **all** options at once, whereas API proposed by PEP 741 sets a single option. They are different API. Moreover, PyConfig_Set() treats most options as read-only: https://peps.python.org/pep-0741/#configuration-options